server/transaction/manager.go

```diff
@@ -59,20 +59,44 @@ type Tx interface {
 	Rollback(ctx context.Context) error
 }
 
+type stagedDatabase interface {
+	IsMetadataChange() bool
+}
+
 // SessionCtx is used to store any baggage for the lifetime of the transaction. We use it to stage the database inside
 // a transaction when the transaction is performing any DDLs.
 type SessionCtx struct {
-	db any
+	db                    stagedDatabase
+	noMetadataChanged     *bool
+	metadataChangeSession bool
 }
 
-func (c *SessionCtx) StageDatabase(db any) {
+func (c *SessionCtx) MetadataChangeSession(mcs bool) {
+	c.metadataChangeSession = mcs
+}
+
+func (c *SessionCtx) StageDatabase(db stagedDatabase) {
 	c.db = db
 }
 
-func (c *SessionCtx) GetStagedDatabase() any {
+func (c *SessionCtx) GetStagedDatabase() stagedDatabase {
 	return c.db
 }
 
+func (c *SessionCtx) MarkNoMetadataStateChanged() {
+	ptrFalse := false
+	c.noMetadataChanged = &ptrFalse
```

Comment:
Why this need to be a pointer?

Reply:
Why we cannot just set c.metadataChangeSession to true here?

Reply:
What will the benefit of setting it to true?